0001-short-term-release-process/0001-release-process-short-term.md

```diff
@@ -0,0 +1,325 @@
+
+* Internal development and testing phase (two weeks):
+    - Begins immediately after production deploy of previous wave
+    - Deploy previous stable version to `STG-02` **
+    - Feature branches are worked on and merged back into `main`
+    - Developers can deploy their own isolated testnets if necessary
+    - We can do quick `alpha` releases if necessary
+* Release candidate (RC) phase (one week):
+    - Create a `release-YYYY.MM.rc.1` branch from `main`. No new features will be accepted on this
+      branch.
+    - Bump version numbers, with an `rc.1` suffix applied.
+    - Build RC and release to Github as a public `pre-release` but with no published crates.
+    - Deploy RC to `STG-01`.
+    - Production of changelog begins.
+    - Invite community members to test against this network.
+    - Shu's metrics solution enables comparisons to be made to the previous stable release in `STG-02`.
+    - Fixes can be applied to the release branch, resulting in an `rc.2`. This will be
+      released/deployed/tested. Repeat if necessary.
+* Release and deploy phase (one/two days):
+    - On the release branch, modify version numbers to remove the `rc` pre-release specifier.
+    - Changelog finalised.
+    - Merge the release branch into `stable`.
+    - Merge the release branch into `main`.
+    - When ready, perform a release and tag of `stable` by kicking off a workflow. *
+    - Delete the release branch.
+    - Deploy to `PROD-01`/`02` and drain *
+    - Announce to users
+
+We can accommodate hotfixes at any point during the cycle.
+
+#### Uncertainties
+
+* Not completely sure about the duration of each phase. Any opinions?
+* We could deploy automatically, but do we need to coordinate with an announcement?
+* What does the 'draining' process with two production environments look like? Shu to elaborate on
+the details? Do we need this on an environment that we upgrade?
+
+## Release Cycle Anatomy
+
+We'll now elaborate the release cycle described in the last section, discussing each type of release
+in more detail.
+
+### Artifacts
+
+All release types will produce a set of binary artifacts, so we'll discuss these first. In Rust,
+crates must use Semantic Versioning. A binary is defined within a crate, and therefore, by default,
+it will also have a Semantic Version; however, it is possible to override the `--version` argument
+on the binary to provide something custom. Our releases currently produce eight binary artifacts. *
+It would be useful if we could refer to these collectively with a single version number and package,
+where the package name would reflect the version number. ** The `--version` argument would identify
+this version number, but also identify the individual component using its Semantic Version. *
+
+The proposal is to use `YYYY.MM.X.Y` as the collective version number. **
+
+We would use the collective version number for a single Github Release. The assets for the release
```

Comment:
Will the crates still have their individual version numbers? that's not clear to me here

Reply:
When it comes to crates, those are obligated to have semantic versions. If we're talking about binaries, I think it would be useful to keep the Semantic Version on each of the individual binaries; however, Shu is of the opinion that the Semantic Version should be removed from everything that's user facing. We need to make a decision on it.

Reply:
My vote on the binaries is to have 2 separate flags, `--version` that shows the collective version as is say `YYYY.MM.X.Y` (easy for the consumer to relate all their artifacts are from the same bundle), and `--protocol-version` that reflects the `sn_networking` crate's semantic versioning (the actual label for the `--protocol-version` can be changed if desired).

Reply:
Is it then something like 

`bundle-version` for a release bundle and `version` for the actual version of a crate and `protocol-version` for the version of the protocol?

Seems like we can have it all, if we're clear about it. 

`bundle version` seems fine and simple as the default user facing output?

Reply:
Thanks for the input on this.

Sorry Josh, but your last statement introduces a bit of uncertainty here for me. I interpreted your first statement as this:
`--version`: outputs the Semantic Version (crate version)
`--bundle-version`: outputs the packaged `YYYY.MM.DD.X.Y` version
`--protocol-version`: outputs the network protocol version

However, you then seem to suggest that `--version` would output the bundle version?

I honestly don't see why we couldn't just have the bundled version and the Semantic Version on the `--version` output? I want to keep the Semantic Version in because I think it's useful to be able to distinguish the individual components and also because `safeup` and `safenode-manager` works on the Semantic Versions. I honestly don't believe our time would be well spent just now on that refactor. I don't see why it's so unreasonable for the user to see the Semantic Version.

I don't want to keep going round in circles on this though. Shu has made his argument and I've made mine. Would you be able to clarify your point on the bundled version and help us reach a conclusion here please?

Reply:
Adding in that for monitoring and grafana dashboards, having multiple versions in a `--version` output would end up storing multiple versions in a single string within a single column in the dashboards since there is no way to separate these out, including the output that comes from `safenode-manager status --json` for version.

My vote is to keep all 3 as separate parameters, so we can get all 3 version into separate columns on the monitoring as well.

Reply:
The decision has to be balanced against the refactoring that would need to be done for `safeup` and `safenode-manager` to use the different version number formatting. I think it's too much of a glaring inconsistency to have those tools use Semantic Versions when those versions are not returned with `--version` in the binaries.

So, I still vote against this, even with the Dashboard impact. Need a decision from Josh here.

Reply:
seems like we can have 

```--package-version: outputs the packaged YYYY.MM.X.Y version
--crate-version: outputs the Semantic Version (crate version)
--protocol-version: outputs the network protocol version``` to get the dashboard on track.

And then the only question is what output do we get/want from `--version`? And is there any reason not to just output it all? (which, if all things can be gathered separately, i don't think there is?)

Reply:
I think the above suggestion is what we should go with.